internal/processing/processors/v2alpha1/virtualservice/virtual_service_processor.go

```diff
@@ -89,7 +95,15 @@ func (r virtualServiceCreator) Create(api *gatewayv2alpha1.APIRule) (*networking
 
 	vsSpecBuilder := builders.VirtualServiceSpec()
 	for _, host := range api.Spec.Hosts {
-		vsSpecBuilder.AddHost(default_domain.GetHostWithDomain(string(*host), r.defaultDomainName))
+		if helpers.IsShortHostName(string(*host)) {
+			if r.gateway == nil || len(r.gateway.Spec.Servers) < 1 || len(r.gateway.Spec.Servers[0].Hosts) < 1 {
+				return nil, errors.New("gateway or host definition is missing")
+			}
+			gatewayDomain := strings.TrimPrefix(r.gateway.Spec.Servers[0].Hosts[0], "*.")
```

Comment:
same as in https://github.com/kyma-project/api-gateway/pull/1311#discussion_r1773283691
If we expect that the host is the first one in the first server, we should check that this is the case (and make it so that the number of servers must be 1)
Or we should get the domain here iteratively

Reply:
did a bit of refactoring here as well, take a look again.